code/game/mecha/working/ripley.dm

```diff
@@ -1,46 +1,79 @@
 /obj/mecha/working/ripley
 	desc = "Autonomous Power Loader Unit. The workhorse of the exosuit world."
-	name = "APLU \"Ripley\""
+	name = "APLU MK-I \"Ripley\""
 	icon_state = "ripley"
 	initial_icon = "ripley"
-	step_in = 4
+	silicon_icon_state = "ripley-empty"
+	step_in = 2.5 //Move speed, lower is faster.
+	/// How fast the mech is in low pressure
+	var/fast_pressure_step_in = 1.5
+	/// How fast the mech is in normal pressure
+	var/slow_pressure_step_in = 2.5
 	max_temperature = 20000
 	health = 200
+	damage_absorption = list("brute"=0.65,"fire"=1.2,"bullet"=0.9,"laser"=1,"energy"=1,"bomb"=1)
 	wreckage = /obj/effect/decal/mecha_wreckage/ripley
+	enclosed = FALSE
+	enter_delay = 2
 	mech_sprites = list(
 		"ripley",
 		"ripley_glass",
-		"titan",
-		"ripley_flames_red",
-		"ripley_flames_blue",
-		"hivisripley"
+		"hauler"
 	)
 	paintable = 1
 
+/obj/mecha/working/ripley/Move()
+	. = ..()
+	update_pressure()
+
 /*
 /obj/mecha/working/ripley/New()
 	..()
 	return
 */
 
-/obj/mecha/working/ripley/firefighter
+/obj/mecha/working/ripley/mk2
+	desc = "Autonomous Power Loader Unit. The workhorse of the exosuit world, this variant is fitted with a pressurized cabin. "
+	name = "APLU MK-II \"Ripley\""
+	icon_state = "ripleymkii"
+	initial_icon = "ripleymkii"
+	step_in = 2.5 //Move speed, lower is faster.
+	/// How fast the mech is in low pressure
+	fast_pressure_step_in = 2.5
+	/// How fast the mech is in normal pressure
+	slow_pressure_step_in = 4
+	health = 250
+	damage_absorption = list("brute"=0.6,"fire"=1.2,"bullet"=0.9,"laser"=1,"energy"=1,"bomb"=0.8)
```

Comment:
In TG's code it has outright better melee, bullet, same laser, and the Gygax/Durand have 0-15% bomb resist, while a Ripley has 60%. Probably because TG has gibonite that explodes when a mech drills it while this server doesn't, so you're right it's probably not necessary here.
That said it.. doesn't have superior bullet/laser resist, the Gygax has 0.8 bullet & 0.7 laser, so it's somewhat worse.

Reply:
Looking into exactly how these are calculated it isn't that bad actually, its a multiplicative factor not a straight reduction.
Fair enough keep the values.